src/EditorFeatures/Core/Diagnostics/DiagnosticsClassificationTaggerProvider.cs

```diff
@@ -62,18 +62,47 @@ protected internal override bool IsEnabled
             => !_editorOptionsService.Factory.GlobalOptions.GetOptionValue(DefaultTextViewHostOptions.IsInContrastModeId);
 
         protected internal override bool IncludeDiagnostic(DiagnosticData data)
-            => data.CustomTags.Contains(WellKnownDiagnosticTags.Unnecessary);
+        {
+            if (!data.CustomTags.Contains(WellKnownDiagnosticTags.Unnecessary))
+            {
+                // All unnecessary code diagnostics should have the 'Unnecessary' custom tag.
+                // Below assert ensures that we do no report unnecessary code diagnostics that
+                // want to fade out multiple locations which are encoded as
+                // additional location indices in the diagnostic's property bag
+                // without the 'Unnecessary' custom tag. 
+                Debug.Assert(!TryGetUnnecessaryLocationIndices(data, out _));
```

Comment:
I expect this to fail with `AbstractUseInferredMemberNameDiagnosticAnalyzer` (the derived C# and VB implementations reports unnecessary locations, but the base `AbstractUseInferredMemberNameDiagnosticAnalyzer` doesn't yet derive from `AbstractBuiltInUnnecessaryCodeStyleDiagnosticAnalyzer`.

If it didn't fail, it would be good to understand why.

Reply:
I see now the last commit 1b2adbd addresses the question :)